flink-runtime/src/main/java/org/apache/flink/runtime/state/StateSnapshotContextSynchronousImpl.java

```diff
@@ -130,7 +136,7 @@ public OperatorStateCheckpointOutputStream getRawOperatorStateOutput() throws Ex
 	}
 
 	private <T extends StreamStateHandle> void closeAndUnregisterStream(
-		NonClosingCheckpointOutputStream<T> stream) throws IOException {
+		NonClosingCheckpointOutputStream<? extends T> stream) throws IOException {
```

Comment:
Should be sufficient to have `NonClosingCheckpointOutputStream<? extends StreamStateHandle>`

Reply:
👍